src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -211,7 +253,7 @@ protected DataContractCriticalHelper Helper
             { return _helper; }
         }
 
-        internal Type UnderlyingType
+        public Type UnderlyingType
```

Comment:
For NetCore I've only changed for internal type so it shouldn't change the public surface. Only in NetNative we have generated files that referencing to these types so they are public.